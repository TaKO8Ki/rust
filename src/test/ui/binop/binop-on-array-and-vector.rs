fn main() {
    struct X;
    let foo = [X, X, X];
    foo == [X, X, X]; //~ ERROR binary operation `==` cannot be applied to type `[X; 3]` [E0369]

    let bar = vec![X, X, X];
    bar == [X, X, X]; //~ ERROR binary operation `==` cannot be applied to type `Vec<X>` [E0369]
}
