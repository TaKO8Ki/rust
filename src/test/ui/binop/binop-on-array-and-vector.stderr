error[E0369]: binary operation `==` cannot be applied to type `[X; 3]`
  --> $DIR/binop-on-array-and-vector.rs:4:9
   |
LL |     foo == [X, X, X];
   |     --- ^^ --------- [X; 3]
   |     |
   |     [X; 3]
   |
note: an implementation of `PartialEq<_>` might be missing for `X`
  --> $DIR/binop-on-array-and-vector.rs:2:5
   |
LL |     struct X;
   |     ^^^^^^^^^ must implement `PartialEq<_>`
help: consider annotating `X` with `#[derive(PartialEq)]`
   |
LL |     #[derive(PartialEq)]
   |

error[E0369]: binary operation `==` cannot be applied to type `Vec<X>`
  --> $DIR/binop-on-array-and-vector.rs:7:9
   |
LL |     bar == [X, X, X];
   |     --- ^^ --------- [X; 3]
   |     |
   |     Vec<X>
   |
note: an implementation of `PartialEq<_>` might be missing for `X`
  --> $DIR/binop-on-array-and-vector.rs:2:5
   |
LL |     struct X;
   |     ^^^^^^^^^ must implement `PartialEq<_>`
help: consider annotating `X` with `#[derive(PartialEq)]`
   |
LL |     #[derive(PartialEq)]
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0369`.
